### DISCUSSION TITLE
How to use a pure-Python loss function

### DISCUSSION THREAD

**Discussion by x66ccff** [MilesCranmer/PySR#517] [Q&A]
Hi Miles,

Thank you for developing this excellent library! 

I read the discussion boards and issues but seem to find that the definition of the loss function is based on Julia. Is there a way for loss functions to call python libraries? Using a pure python function to evaluate the fitness of an expression?

I think this is a very important feature, which can make PySR more extensible, I hope you can help me!

Looking forward to your reply.

**Comment by MilesCranmer** [comment]
Hi @x66ccff,

This might be possible with PyCall.jl (calling Python from within Julia), but I highly recommend against it. The evaluation function needs to be very fast and any python code is going to slow things down. What sort of Python function do you have?

Cheers,
Miles
